server/src/test/java/org/elasticsearch/snapshots/SnapshotInfoXContentTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.snapshots;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.PlainActionFuture;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
+import org.elasticsearch.test.AbstractWireTestCase;
+
+import java.io.IOException;
+
+public class SnapshotInfoXContentTests extends AbstractWireTestCase<SnapshotInfo> {
```

Comment:
It seems that SnapshotInfo is lenient when it comes to parse unknown fields. Should we add some AbstractXContentTestCase just for SnapshotInfo to test this too?

Reply:
Unfortunately we don't seem to parse it as `AbstractXContentTestCase` requires today. Likely fixable but I didn't really want to make production code changes here. At least we have good tests for the behaviour of repositories after upgrades and downgrades which should catch any problems in this area.

Reply:
Sure, let's move on.

Reply:
It turns out we do have some coverage of this area via `GetSnapshotsResponseTests`; improvements are coming in https://github.com/elastic/elasticsearch/pull/72079.